crates/nargo_cli/src/cli/codegen_verifier_cmd.rs

```diff
@@ -19,8 +21,22 @@ pub(crate) fn run(args: CodegenVerifierCommand, config: NargoConfig) -> Result<(
     #[allow(deprecated)]
     let smart_contract_string = backend.eth_contract_from_cs(compiled_program.circuit);
 
-    let contract_dir = config.program_dir.join(CONTRACT_DIR);
-    create_named_dir(&contract_dir, "contract");
+    let manifest_path = config.program_dir.join(PKG_FILE);
+    if !manifest_path.exists() {
+        return Err(CliError::MissingTomlFile("Nargo".to_owned(), manifest_path));
+    }
+
+    let manifest = match manifest::parse(manifest_path) {
+        Ok(it) => it,
+        Err(err) => return Err(CliError::Generic(err.to_string())),
+    };
+
+    let contract_dirname = match manifest.codegen_verifier {
+        Some(it) => it.out,
+        None => "verifier".to_owned(),
+    };
+    let contract_dir = config.program_dir.join(contract_dirname.clone());
```

Comment:
Let's define this default value logic inside of `manifest/mod.rs`. I think a nice flow for now would be to have a `pub(crate)` function on `PackageManifest` which would return the value of `contract_dir` here (defaulting to `PKG_ROOT/verifier` if it's unset).

Reply:
Ah, sorry I forgot that `PackageManifest` doesn't track the root directory itself, let's just wrap this into a small `verifier_output_dir` function which lives in this file.

I got mixed up with a future feature I want to add but doesn't exist just yet.

Reply:
Amended. https://github.com/noir-lang/noir/pull/1138/commits/31577ac5c0d2aa6156ee2dbcadb3969f85bbd906